src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,32 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj instanceof Rectangle) {
+            Rectangle rectangleObj = (Rectangle) obj;
+            if (!rectangleObj.getColor().equals(rectangleObj.getColor())) {
+                return false;
+            }
+            if (rectangleObj.getLength() != rectangleObj.getLength()) {
+                return false;
+            }
+            if (rectangleObj.getWidth() != rectangleObj.getWidth()) {
+                return false;
+            }
+            return true;
+        }
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = width * length;
+        result = 31 * result + (color != null ? color.length() : 0);
```

Comment:
try to make your hashcode more complex, to avoid collisions. hint: you can use hashcodes of Rectangle's fields